roles/pulp_common/defaults/main.yml

```diff
@@ -11,11 +11,7 @@ __pulp_daemons_dir: '{{ (pulp_install_source == "packages") | ternary("/usr/libe
 pulp_install_api_service: true
 # Deprecated unused. Variables for dependency upgrades are TBD
 pulp_upgrade: false
-#  Intentionally not advertised to users. They should not set this unless
-#  they are confident that this version of pulp_installer is compatible with
-#  pulp_version.
-#  Ignored if pulp_source_dir is set
-pulp_version: "3.8.1"
+__pulp_version: '{{ pulp_version | default("3.8.1") }}'
```

Comment:
I believe, this should move to vars.
Also if i read it correctly, it is only used for pip installs. It should maybe be called `__pulp_common_pulpcore_pip_version`?

Reply:
Agree on moving it to the `vars/main.yml` file.

For the rename, I tend to lean toward keeping `__pulp_version`. It is generic and accurate enough that this value can be leveraged for other purposed that is not `pip` related specifically.

Reply:
It is only used for pip installations. And for anything else i guess one should use `pulp_version`.
Ideally we didn't even default_pin the python version in the installer and just install latest from pypi or the specified version.

Reply:
@mikedep333 @fao89 @dkliban what are your thoughts on that one ?

Reply:
@Spredzy : @mdellweg and I discussed this last meeting. There were multiple problems that pinning the version solved. This one in particular seemed to be the most tricky to solve.

>  4. Handling the case where a user may run an outdated installer to install the current version of Pulp, and thus either the install failing or pulp failing at runtime because of it.

https://pulp.plan.io/issues/5890

I am open to alternate approaches, but we should review them based on the discussions in that issue and the feedback in the [initial design PR (that was centered around branches).](https://github.com/pulp/pulp_installer/pull/203)